snow/consensus/snowman/poll/early_term_no_traversal.go

```diff
@@ -67,20 +177,39 @@ func (p *earlyTermNoTraversalPoll) Drop(vdr ids.NodeID) {
 //     transitive voting.
 //  4. A single element has achieved an alphaConfidence majority.
 func (p *earlyTermNoTraversalPoll) Finished() bool {
+	if p.finished {
+		return true
+	}
+
 	remaining := p.polled.Len()
 	if remaining == 0 {
+		p.finished = true
+		p.metrics.observeExhausted(time.Since(p.start))
 		return true // Case 1
 	}
 
 	received := p.votes.Len()
 	maxPossibleVotes := received + remaining
 	if maxPossibleVotes < p.alphaPreference {
+		p.finished = true
+		p.metrics.observeEarlyFail(time.Since(p.start))
 		return true // Case 2
 	}
 
 	_, freq := p.votes.Mode()
-	return freq >= p.alphaPreference && maxPossibleVotes < p.alphaConfidence || // Case 3
-		freq >= p.alphaConfidence // Case 4
+	if freq >= p.alphaPreference && maxPossibleVotes < p.alphaConfidence {
```

Comment:
They do the same thing. They're separated here solely to differentiate between them in the metrics. Currently, `alphaPreference == alphaConfidence` by default, which means case 3 can never be triggered.

Imo it's useful to differentiate these for gathering metrics with different consensus parameters, but there's no short term plan to change the defaults, so we could re-combine these if desired. I'd prefer to leave as is, since there's an important difference between these two cases.

I could add a comment explaining that case 3 and case 4 are handled the same with the exception of the metrics though if you think that would be helpful.

Reply:
I'm not sure I understand this comment.

If
- `freq >= p.alphaPreference` is `true`
- `freq >= p.alphaConfidence` is `false`
- `maxPossibleVotes < p.alphaConfidence` is `false`

the function returns `false`.

---

If
- `freq >= p.alphaPreference` is `true`
- `freq >= p.alphaConfidence` is `false`
- `maxPossibleVotes < p.alphaConfidence` is `true`

the function returns `true`.

So the result of `maxPossibleVotes < p.alphaConfidence` is critical for the evaluation of the function.